contracts/tobin_tax.go

```diff
@@ -0,0 +1,53 @@
+package contracts
```

Comment:
It would be good to have a package doc for this package. At the moment it looks like a bit of a mishmash and its not really clear to me what is the functionality it encapsulates.

Looking at the public api from the godoc it looks like splitting it into 2 packages could be a good idea, a tobin package and a different package, I'm not really sure that contracts makes sense as a package for these system call functions, but then again without package doc its hard to understand the purpose.
```
func CallFromSystem(evm *vm.EVM, contractAddress common.Address, abi abipkg.ABI, funcName string, args []interface{}, returnObj interface{}, gas uint64, value *big.Int) (uint64, error)
func ComputeTobinTax(evm *vm.EVM, sender common.Address, transferAmount *big.Int) (tax *big.Int, taxRecipient common.Address, err error)
func GetRegisteredAddress(evm *vm.EVM, registryId common.Hash) (common.Address, error)
func StaticCallFromSystem(evm *vm.EVM, contractAddress common.Address, abi abipkg.ABI, funcName string, args []interface{}, returnObj interface{}, gas uint64) (uint64, error)
func TobinTax(evm *vm.EVM, sender common.Address) (numerator *big.Int, denominator *big.Int, reserveAddress common.Address, err error)
```

Reply:
I think it can be split into 1) system calls / EVM communication
```
CallFromSystem(....)
StaticCallFromSystem(...)
GetRegisteredAddress(...) // this is a more involved static call to system
```
and 2) Tobin Tax
```
ComputeTobinTax(...) // don't need TobinTax to be public
```

Reply:
I agree that tobin tax belongs to its own module, and imo same applies to registry access (but there's a circular dependency there)

tobin tax code also uses a hardcoded way of doing a contract call; that would be nice to modify

I didn't do those changes to avoid making the PR bigger. But i can at least move tobin tax for now

Reply:
system call may not be the best name, but but one definition is a call not associated with an externally owned account (i.e. systemcaller address is `0x0`). The static calls are read only calls to some notion of the current state. The `CallFromSystem` modifies the provided state as part of the consensus protocol (like paying out validator rewards or updating the gas price minimum)